platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/DataFileCompactor.java

```diff
@@ -438,6 +442,18 @@ public boolean compact() throws IOException, InterruptedException {
                     (filesToCompactSize - compactedFilesSize) * UnitConstants.BYTES_TO_MEBIBYTES);
         }
 
+        if (reportTotalSpaceMetricFunction != null) {
+            Map<Integer, List<DataFileReader<?>>> readersByLevel =
+                    getReadersByLevel(dataFileCollection.getAllCompletedFiles());
+            for (int i = 0; i < MAX_COMPACTION_LEVEL; i++) {
```

Comment:
Why not use the actual number of entries in the map instead of the maximum compaction level?

Reply:
For no particular reason. You're right, map size as a right boundary is more appropriate.